### PR TITLE
DOCS-5553 Adds Static Analysis Doc Set + Single Sourcing Logic for GitHub Actions and CircleCI Orb Documentation

### PR DESCRIPTION
**Pull request by alai97**
### What does this PR do?

Adds a section for Static Analysis in the CI Visibility doc set (one manual page, two single-sourced pages).

### Motivation

DOCS-5553

### Additional Notes

Remember to update the `pull_config_preview.yaml` and `pull_config.yaml` files before merging!

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [x] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/continuous_integration/static_analysis/_index.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+The [`@datadog/datadog-ci` package][2] allows you to run Static Analysis directly within your CI/CD pipeline. To use the [`@datadog/datadog-ci` NPM package][2], see [Configuration][3].
+
+## Further Reading
+
+{{< partial name="whats-next/whats-next.html" >}}
+
+[1]: https://app.datadoghq.com/ci/static-analysis
+[2]: https://www.npmjs.com/package/@datadog/datadog-ci
+[3]: /static_analysis/configuration/
```
Comment (alai97) [inline content/en/continuous_integration/static_analysis/_index.md:58]:
```suggestion
[3]: /continuous_integration/static_analysis/configuration/
```

File: local/bin/py/build/configurations/pull_config_preview.yaml [hunk 2]
```diff
@@ -334,6 +334,36 @@
...
+          globs:
+          - 'README.md'
+          options:
+            dest_path: '/continuous_integration/static_analysis/'
+            file_name: 'github_actions.md'
+            front_matters:
+              dependencies: ["https://github.com/DataDog/datadog-static-analyzer-github-action/blob/main/README.md"]
+              title: Static Analysis and GitHub Actions
+              kind: documentation
+              description: Use the Datadog Static Analysis CI extension to run Static Analysis tests in a CI pipeline.
```
Comment (alai97) [inline local/bin/py/build/configurations/pull_config_preview.yaml:365]:
```suggestion
              description: Use Datadog and GitHub to run Static Analysis jobs in a CI pipeline.
```

File: local/bin/py/build/configurations/pull_config_preview.yaml [hunk 3]
```diff
@@ -334,6 +334,36 @@
...
+          globs:
+          - 'README.md'
+          options:
+            dest_path: '/continuous_integration/static_analysis/'
+            file_name: 'circleci_orb.md'
+            front_matters:
+              dependencies: ["https://github.com/DataDog/datadog-static-analyzer-circleci-orb/blob/main/README.md"]
+              title: Static Analysis and CircleCI Orb
+              kind: documentation
+              description: Use the Datadog Static Analysis CI extension to run Static Analysis tests in a CI pipeline.
```
Comment (alai97) [inline local/bin/py/build/configurations/pull_config_preview.yaml:350]:
```suggestion
              description: Use Datadog and CircleCI to run Static Analysis jobs in a CI pipeline.
```

File: content/en/continuous_integration/static_analysis/_index.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+* Source Code: A description of the violation and the offending lines of code that caused it. 
+
+  To see the offending code snippet, configure the [Datadog GitHub App][4].
+  
+  {{< img src="monitors/incidents/TODO.png" alt="Static Analysis result Source Code Tab" style="width:80%;">}}
+
+* Fix: Where possible, Datadog recommends one or more code fixes to resolve the violation that a developer can copy and paste.
+
+  {{< img src="monitors/incidents/TODO.png" alt="Static Analysis result Fix Tab" style="width:80%;">}}
```
Comment (alai97) [inline content/en/continuous_integration/static_analysis/_index.md:72]: Screenshot to be added

```suggestion
  {{< img src="monitors/incidents/TODO.png" alt="Static Analysis result Fix Tab" style="width:80%;">}}
```

File: content/en/continuous_integration/static_analysis/_index.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+Click on a violation to open a side panel containing metadata regarding the scope of the violation and where it originated from. 
+
+The content of the violation is split into three tabs:
+
+* Source Code: A description of the violation and the offending lines of code that caused it. 
+
+  To see the offending code snippet, configure the [Datadog GitHub App][4].
+  
+  {{< img src="monitors/incidents/TODO.png" alt="Static Analysis result Source Code Tab" style="width:80%;">}}
```
Comment (alai97) [inline content/en/continuous_integration/static_analysis/_index.md:68]: Screenshot to be added

```suggestion
  {{< img src="monitors/incidents/TODO.png" alt="Static Analysis result Source Code Tab" style="width:80%;">}}
```

File: content/en/continuous_integration/static_analysis/_index.md [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+Datadog Static Analysis is run in your CI pipelines using the [datadog-ci CLI][2] to check your code against Datadog's default rulesets. To use the [`@datadog/datadog-ci` NPM package][2], see [Configuration][3].
+
+### Search and filter results
+
+After configuring your CI pipelines to run the Datadog Static Analyzer, violations will appear on the [Static Analysis Results page][1]. To filter your results, use the facets to the left of the list or use the search bar. 
+
+Each violation is associated with a specific commit and branch from your repository on which your CI pipeline was run. Every violation per commit has a row item in the list. 
+
+{{< img src="monitors/incidents/TODO.png" alt="Static Analysis results list view" style="width:80%;">}}
```
Comment (alai97) [inline content/en/continuous_integration/static_analysis/_index.md:58]: Screenshot to be added

```suggestion
{{< img src="monitors/incidents/TODO.png" alt="Static Analysis results list view" style="width:80%;">}}
```

File: content/en/continuous_integration/static_analysis/configuration.md [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+```
+
+[101]: /account_management/api-app-keys/#api-keys
+[102]: /account_management/api-app-keys/#application-keys
+[103]: /getting_started/site/
+
+{{% /tab %}}
+{{< /tabs >}}
+
+### Import third-party Static Analysis results
```
Comment (kayayarai) [inline content/en/continuous_integration/static_analysis/configuration.md:117]:
```suggestion
### Upload third-party static analysis results to Datadog
```

File: content/en/continuous_integration/static_analysis/configuration.md [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+
+Datadog supports ingesting Static Analysis results from third-party Static Analysis tools that can export results into the interoperable [Static Analysis Results Interchange Format (SARIF) Format][3].
```
Comment (kayayarai) [inline content/en/continuous_integration/static_analysis/configuration.md:119]:
```suggestion
You can send results from third-party static analysis tools to Datadog, provided they are in the interoperable [Static Analysis Results Interchange Format (SARIF) Format][3]. 
```

File: content/en/continuous_integration/static_analysis/configuration.md [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+To upload a SARIF report:
+
+1. Ensure the [`DD_API_KEY` and `DD_APP_KEY` variables are defined][2].
+2. Install the `datadog-ci` utility:
+   
+   ```bash
+   npm install -g @datadog/datadog-ci
+   ```
+
+3. Run your third-party Static Analysis tool on your code which outputs results in the SARIF format.
```
Comment (kayayarai) [inline content/en/continuous_integration/static_analysis/configuration.md:130]:
```suggestion
3. Run the third-party static analysis tool on your code and output the results in the SARIF format.
```

File: content/en/continuous_integration/static_analysis/configuration.md [hunk 10]
```diff
@@ -0,0 +1,143 @@
...
+4. Upload the results to Datadog:
+
+   ```bash
+   datadog-ci sarif upload $OUTPUT_LOCATION --service <datadog-service> --env <datadog-env>
```
Comment (kayayarai) [inline content/en/continuous_integration/static_analysis/configuration.md:134]:
```suggestion
   datadog-ci sarif upload $OUTPUT_LOCATION --service <datadog-service> --env <datadog-env> --site <dd-site>
```

**Review by kayayarai — APPROVED** [review]

**Comment by kayayarai** [comment]
Noting: Still to do: updates to pull_config.yaml and add screen shots

File: config/_default/menus/menus.en.yaml [hunk 11]
```diff
@@ -2167,16 +2167,31 @@ main:
...
     weight: 306
+  - name: Static Analysis
+    url: continuous_integration/static_analysis/
+    parent: ci
+    identifier: ci_static_analysis
+    weight: 4
+  - name: CircleCI Orb
+    url: continuous_integration/static_analysis/circleci_orb
+    parent: ci_static_analysis
+    identifier: ci_static_analysis_circleci_orb
```
Comment (alai97) [inline config/_default/menus/menus.en.yaml:2178]:
```suggestion
  - name: CircleCI Orbs
    url: continuous_integration/static_analysis/circleci_orbs
    parent: ci_static_analysis
    identifier: ci_static_analysis_circleci_orbs
```

File: local/bin/py/build/configurations/pull_config_preview.yaml [hunk 12]
```diff
@@ -334,6 +334,36 @@
...
+        - action: pull-and-push-file
+          branch: main
+          globs:
+          - 'README.md'
+          options:
+            dest_path: '/continuous_integration/static_analysis/'
+            file_name: 'circleci_orb.md'
+            front_matters:
+              dependencies: ["https://github.com/DataDog/datadog-static-analyzer-circleci-orb/blob/main/README.md"]
+              title: Static Analysis and CircleCI Orb
```
Comment (alai97) [inline local/bin/py/build/configurations/pull_config_preview.yaml:348]:
```suggestion
            file_name: 'circleci_orbs.md'
            front_matters:
              dependencies: ["https://github.com/DataDog/datadog-static-analyzer-circleci-orb/blob/main/README.md"]
              title: Static Analysis and CircleCI Orbs
```

File: content/en/continuous_integration/static_analysis/_index.md [hunk 13]
```diff
@@ -0,0 +1,173 @@
...
+
+Using Static Analysis provides organizations with the following benefits:
+
+* Static Analysis takes the guesswork out of adhering to an organization's code standards, enabling your development team to ship compliant code without significant impacts to developer velocity.
+* New developers to an organization are able to onboard faster because Static Analysis enables an organization to maintain a more readable codebase over time.
+* An organization's software becomes reliable over time by virtue of the code being more maintainable because the risk of a developer introducing new defects to the code is minimized.
+
+## Integrations
+
+{{< whatsnext desc="With Static Analysis, you can integrate feedback in a supporting CI platform provider, with native integrations for GitHub and CircleCI. See the documentation for information about the following integrations:">}}
```
Comment (alai97) [inline content/en/continuous_integration/static_analysis/_index.md:39]:
```suggestion
{{< whatsnext desc="With Static Analysis, you can integrate feedback on code reviews for various languages in any CI platform provider of choice. See the documentation for information about the following integrations:">}}
```

File: local/bin/py/build/configurations/pull_config_preview.yaml [hunk 14]
```diff
@@ -1,6 +1,6 @@
 ---
 - config:
-    cache_enabled: true
+    cache_enabled: false
```
Comment (alai97) [inline local/bin/py/build/configurations/pull_config_preview.yaml:3]:
```suggestion
    cache_enabled: true
```

**Pull request merged**